packages/@aws-cdk/aws-backup/lib/rule.ts

```diff
@@ -185,6 +216,20 @@ export class BackupPlanRule {
       throw new Error(`'deleteAfter' must be between 1 and 35 days if 'enableContinuousBackup' is enabled, but got ${props.deleteAfter.toHumanString()}`);
     }
 
+    if (Array.isArray(props.copyActions) && props.copyActions.length > 0) {
+      props.copyActions.forEach(copyAction => {
+        if (copyAction.deleteAfter && copyAction.moveToColdStorageAfter &&
+          copyAction.deleteAfter.toDays() < copyAction.moveToColdStorageAfter.toDays()) {
+          throw new Error('`deleteAfter` in copyActions must be greater than corresponding `moveToColdStorageAfter`');
+        }
+
+        if (copyAction.deleteAfter && copyAction.moveToColdStorageAfter &&
+          copyAction.deleteAfter.toDays() < copyAction.moveToColdStorageAfter.toDays() + 90) {
+          throw new Error('`deleteAfter` in copyActions must at least 90 days later than corresponding `moveToColdStorageAfter`');
+        }
```

Comment:
```suggestion
        if (
          copyAction.deleteAfter && !Token.isUnresolved(copyAction.deleteAfter)
          copyAction.moveToColdStorageAfter && !Token.isUnresolved(copyAction.moveToColdStorageAfter)
          copyAction.deleteAfter.toDays() < copyAction.moveToColdStorageAfter.toDays() + 90,
        ) {
          throw new Error([
            '\'deleteAfter\' must at least 90 days later than corresponding \'moveToColdStorageAfter\'',
            `received 'deleteAfter: ${copyAction.deleteAfter.toDays()}' and 'moveToColdStorageAfter: ${copyAction.moveToColdStorageAfter.toDays()}'`,
          ].join('\n'));
        }
```

Reply:
Thanks, this is much more concise now and having the additional feedback in the error message should be very helpful. I've updated this and the corresponding tests.